RecoTracker/TkDetLayers/src/Phase2EndcapRingBuilder.cc

```diff
@@ -0,0 +1,100 @@
+#include "Phase2EndcapRingBuilder.h"
+
+using namespace edm;
+using namespace std;
+
+Phase2EndcapRing* Phase2EndcapRingBuilder::build(const GeometricDet* aPhase2EndcapRing,
+			 			     const TrackerGeometry* theGeomDetGeometry,
+			 			     const bool useBrothers)
+{
+  vector<const GeometricDet*>  allGeometricDets = aPhase2EndcapRing->components();
+  vector<const GeometricDet*>  compGeometricDets;
+  LogDebug("TkDetLayers") << "Phase2EndcapRingBuilder with #Modules: " << allGeometricDets.size() << std::endl;
+
+  vector<const GeomDet*> frontGeomDets;
+  vector<const GeomDet*> backGeomDets;
+  double meanZ = 0;
+
+  if(!useBrothers){
+  
+    //---- to evaluate meanZ
+    for(vector<const GeometricDet*>::const_iterator compGeometricDets=allGeometricDets.begin(); compGeometricDets!=allGeometricDets.end();compGeometricDets++){
+        LogTrace("TkDetLayers") << " compGeometricDets->positionBounds().perp() " << (*compGeometricDets)->positionBounds().z() << std::endl;
+        meanZ = meanZ + (*compGeometricDets)->positionBounds().z();
+    }
+    meanZ = meanZ/allGeometricDets.size();
+    LogDebug("TkDetLayers") << " meanZ " << meanZ << std::endl;
+  
+    for(vector<const GeometricDet*>::const_iterator compGeometricDets=allGeometricDets.begin();
+        compGeometricDets!=allGeometricDets.end();compGeometricDets++){
+      const GeomDet* theGeomDet = theGeomDetGeometry->idToDet( (*compGeometricDets)->geographicalID() );
+  
+      if( fabs( (*compGeometricDets)->positionBounds().z() ) < fabs(meanZ))
+        frontGeomDets.push_back(theGeomDet);
+  
+      if( fabs( (*compGeometricDets)->positionBounds().z() ) > fabs(meanZ))
+        backGeomDets.push_back(theGeomDet);      
```

Comment:
is equality forbidden or is it to be ignored?
For the former case, maybe there should be an assert or an exception.